crypto/multisig/threshold_multisig_test.go

```diff
@@ -0,0 +1,85 @@
+package multisig
+
+import (
+	"math/rand"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/tendermint/tendermint/crypto"
+	"github.com/tendermint/tendermint/crypto/ed25519"
+	"github.com/tendermint/tendermint/crypto/secp256k1"
+)
+
+func TestThresholdMultisig(t *testing.T) {
```

Comment:
I think this test is ripe for table-driven testing.

We should be testing multiple values of N, and make sure we get expected behaviour when N<=0 and when N=1.

We should be testing multiple values of K, including that we get expected behaviour in the edges, eg. when K<=0, K=1, K=N-1, K=N, K>N

Reply:
Same with the other tests really ...

Reply:
We should also test what happens with recursive multisig pubkeys. Or if we want to restrict it, then test it fails.

Reply:
We should also test unmarshalling

Reply:
Table driven sounds good. I do think we should enable recursive multisigs. 

Not really sure why we would test unmarshalling. We assume that amino works, and all variables are public?

Reply:
Right